Signal/src/AppDelegate.m

```diff
@@ -355,6 +356,11 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
     [OWSAnalytics appLaunchDidBegin];
 
+    self.shortcutItem = launchOptions[UIApplicationLaunchOptionsShortcutItemKey];
```

Comment:
We shouldn't need to look at the shortcutItem during `didFinishLaunchingWithOptions`. Even from a cold start, `application:performActionForShortcutItem:completionHandler:` is called. 

The reason the new conversation view isn't presented is because `runNowOrWhenAppDidBecomeReady` will run right after we call `setAppIsReady`, *before* the split view controller is setup. So when we call `[SignalApp.sharedApp showNewConversationView]` `self.conversationSplitViewController` is nil.

Your change is on the right track – we need to hold onto the shortcutItem if `ensureRootViewController` has yet to succeed. Can you refactor this to use `application:performActionForShortcutItem:completionHandler`?